src/rkopenmdao/runge_kutta_integrator.py

```diff
@@ -331,6 +355,15 @@ def _add_inputs_and_outputs(self):
                     shape=quantity.array_metadata.shape,
                     distributed=quantity.array_metadata.distributed,
                 )
+            if quantity.type == "independent_input":
```

Comment:
This very long `if` cascade is not particularly nice. I wonder whether it would be better to have separate classes per quantity type which had specific `add_self(rk_integrator)` method which would encapsulate the `add_in/output()` calls.

Reply:
I think having different classes for the different quantity types would be a good idea, but I'd prefer to not have the quantities depend on the `RungeKuttaIntegrator`. I propose instead to separate the list of quantities in the `TimeIntegrationMetadataClass` into one list per quantity type. With this, it is also possible to get rid of the `if` cascade by introducing one function per list of quantity types that deals with the specifics of that type. Also, this has a further upside: In most cases, when iterating over the list of quantities, only a specific type gets used. When there are lists per type, we can iterate only over the types that get used.